src/DataValues/UnboundedQuantityValue.php

```diff
@@ -0,0 +1,334 @@
+			throw new IllegalValueException( '$unit can not be an empty string (use "1" for unit-less quantities)' );
+		}
+
+		$this->amount = $amount;
+		$this->unit = $unit;
+	}
+
+	/**
+	 * Returns a QuantityValue representing the given amount.
+	 *
+	 * This is a convenience wrapper around the constructor that accepts native values
+	 * instead of DecimalValue objects.
+	 *
+	 * @note: if the amount or a bound is given as a string, the string must conform
+	 * to the rules defined by @see DecimalValue.
+	 *
+	 * @since 0.1
+	 *
+	 * @param string|int|float|DecimalValue $number
+	 * @param string $unit A unit identifier. Must not be empty, use "1" for unit-less quantities.
+	 *
+	 * @return UnboundedQuantityValue
+	 * @throws IllegalValueException
+	 */
+	public static function newFromNumber( $number, $unit = '1' ) {
+		$number = self::asDecimalValue( 'amount', $number );
+
+		return new self( $number, $unit );
+	}
+
+	/**
+	 * @see newFromNumber
+	 *
+	 * @deprecated since 0.1, use newFromNumber instead
+	 *
+	 * @param string|int|float|DecimalValue $number
+	 * @param string $unit
+	 * @param string|int|float|DecimalValue|null $upperBound
+	 * @param string|int|float|DecimalValue|null $lowerBound
+	 *
+	 * @return UnboundedQuantityValue
+	 */
+	public static function newFromDecimal( $number, $unit = '1' ) {
+		return self::newFromNumber( $number, $unit );
+	}
+
+	/**
+	 * Converts $number to a DecimalValue if possible and necessary.
+	 *
+	 * @note: if the $number is given as a string, it must conform to the rules
+	 *        defined by @see DecimalValue.
+	 *
+	 * @param string $name The variable name to use in exception messages
+	 * @param string|int|float|DecimalValue|null $number
+	 * @param DecimalValue|null $default
+	 *
+	 * @throws IllegalValueException
+	 * @throws InvalidArgumentException
+	 * @return DecimalValue
+	 */
+	protected static function asDecimalValue( $name, $number, DecimalValue $default = null ) {
+		if ( !is_string( $name ) ) {
+			throw new InvalidArgumentException( '$name must be a string' );
+		}
+
+		if ( $number === null ) {
+			if ( $default === null ) {
+				throw new InvalidArgumentException( '$' . $name . ' must not be null' );
+			}
+
+			$number = $default;
+		}
+
+		if ( $number instanceof DecimalValue ) {
+			// nothing to do
+		} elseif ( is_int( $number ) || is_float( $number ) || is_string( $number ) ) {
+			$number = new DecimalValue( $number );
+		} else {
+			throw new IllegalValueException( '$' . $name . '  must be a string, int, or float' );
+		}
+
+		return $number;
+	}
+
+	/**
+	 * @see Serializable::serialize
+	 *
+	 * @since 0.1
+	 *
+	 * @return string
+	 */
+	public function serialize() {
+		return serialize( array_values( $this->getArrayValue() ) );
+	}
+
+	/**
+	 * @see Serializable::unserialize
+	 *
+	 * @since 0.1
+	 *
+	 * @param string $data
+	 */
+	public function unserialize( $data ) {
+		list( $amount, $unit ) = unserialize( $data );
+		$amount = DecimalValue::newFromArray( $amount );
```

Comment:
Again, I wonder why you introduced this. Probably for type safety and early exceptions, but the `__constructor` and `asDecimalValue` already do this.